src/reducers/project-state.js

```diff
@@ -435,10 +435,21 @@ const onLoadedProject = (loadingState, canSave, success) => {
         default:
             return;
         }
+    } else {
+        switch (loadingState) {
+        case LoadingState.LOADING_VM_WITH_ID:
+        case LoadingState.LOADING_VM_FILE_UPLOAD:
+            return {
+                type: RETURN_TO_SHOWING
```

Comment:
What is the difference between RETURN_TO_SHOWING and START_ERROR? Since I am not familiar w/the project state machine, I think it would be helpful to have comments describing the state changes that are happening (this does not necessarily need to be added to this PR but would be useful to have in the future).

Reply:
Thanks -- I'll put in a bit of commenting here, and I have done some experiments on how to add documentation to this file that I'd like to show you